cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/query/QueryResult.java

```diff
@@ -122,10 +122,16 @@ public DocumentRevision next() {
             if (!this.hasNext()) {
                 throw new NoSuchElementException();
             }
-            if (!documentBlock.hasNext()) {
+
+            DocumentRevision doc =  documentBlock.next();
```

Comment:
I think this could do with a comment explaining why it is safe to call this without an explicit proceeding `documentBlock.hasNext()` just so it is easier for the next person who wanders into this block.

`this.hasNext()` calls `documentBlock.hasNext()`, but it ORs that result with another check so it is at first glance possible to reach this line without a document being available in the `documentBlock`. I think it would be helpful to clarify that in the case of the initial or final block being empty the range check in `hasNext` protects this call.